kukkuu/oidc.py

```diff
@@ -0,0 +1,148 @@
+import logging
+from typing import Optional
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from helusers.authz import UserAuthorization
+from helusers.jwt import JWT, ValidationError
+from helusers.oidc import AuthenticationError, RequestJWTAuthentication
+from helusers.settings import api_token_auth_settings
+from helusers.user_utils import get_or_create_user
+from jose import jwt as jose_jwt
+
+from kukkuu.tests.utils.jwt_utils import is_valid_256_bit_key
+
+logger = logging.getLogger(__name__)
+
+
+class ApiTokenAuthSettings:
+    def __init__(self, **entries):
+        self.__dict__.update(entries)
+
+
+class BrowserTestAwareJWTAuthentication(RequestJWTAuthentication):
+    def __init__(self):
+        super().__init__()
+        self._api_token_auth_settings = ApiTokenAuthSettings(
+            **{
+                **api_token_auth_settings._settings,
+                **settings.OIDC_BROWSER_TEST_API_TOKEN_AUTH,
+            }
+        )
+        self.algorithms = ["HS256"]
+
+        if self._api_token_auth_settings.ENABLED:
+            if not self._api_token_auth_settings.ISSUER:
+                raise ImproperlyConfigured(
+                    "ISSUER must be configured when test JWT auth is enabled."
+                )
+            if not is_valid_256_bit_key(
+                self._api_token_auth_settings.JWT_SIGN_SECRET
+            ):  # Each byte is 8 bits
```

Comment:
`# Each byte is 8 bits` is true but is it useful information here? The key is expected to be in hexadecimal string format which is probably utf-8-encoded, so each hexadecimal digit (i.e. 4 bits) takes up a single 8-bit character. IMHO the comment is not needed, if you want to make it clearer perhaps adding better documentation to `is_valid_256_bit_key` would be good.